sdgym/benchmark.py

```diff
@@ -442,12 +443,130 @@ def _get_empty_dataframe(compute_quality_score, sdmetrics):
     return scores
 
 
+def _create_sdgym_script(params, output_filepath):
+    first_slash_index = output_filepath.find('/')  # Find the index of the first slash
+    bucket_name = ''
+    key_name = ''
+    if first_slash_index != -1:  # Check if a slash is found
+        bucket_name = output_filepath[:first_slash_index]  # Extract directory part
+        key_name = output_filepath[first_slash_index + 1:]  # Extract filename part
+    else:
+        raise ValueError("""Invalid output_filepath.
+                   The path should be structured as: <s3_bucket_name>/<path_to_file>
+                   Please make sure the path exists and permissions are given.""")
```

Comment:
Can this be put into a utility function and changed to the following? It is currently a bit hard to understand what is going on, and I would suggest something like this: 
```python
    from botocore.exceptions import ClientError
    import boto3

    def parse_s3_path(s3_path):
        if '/' not in s3_path:
            raise ValueError("Invalid S3 path format. Expected 'bucket_name/path_to_file'.")
        bucket_name, key_name = s3_path.split('/', 1)  # Split only on the first '/'
        return bucket_name, key_name  

    def directory_exists(bucket_name, directory_prefix):
        s3_client = boto3.client('s3')
        response = s3.list_objects_v2(Bucket=bucket_name, Prefix=directory_prefix, Delimiter='/')
        return 'Contents' in response or 'CommonPrefixes' in response

    def check_read_permissions(bucket_name):
        s3 = boto3.client('s3')
        try:
            s3.list_objects_v2(Bucket=bucket_name, MaxKeys=1)
            return True  # No exception means we have read permission
        except ClientError as e:
            if e.response['Error']['Code'] in ('AccessDenied', 'AllAccessDisabled'):
                return False  # Access denied or bucket does not exist
            raise  # Re-raise exception if it's something else we didn't expect
```
- You might be able to re-use the boto3.client instead of re-creating it every time like I did.